activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -137,7 +137,7 @@ def commit_records
             record.committed!
           else
             # if not running callbacks, only adds the record to the parent transaction
-            record.add_to_transaction
+            record.class.connection.add_transaction_record(record)
```

Comment:
How about just `connection.add_transaction_record(record)`?

```suggestion
            connection.add_transaction_record(record)
```